libs/zopengl/src/wrapper.zig

```diff
@@ -0,0 +1,2637 @@
+pub const FILL = bindings.FILL;
+pub const KEEP = bindings.KEEP;
+pub const REPLACE = bindings.REPLACE;
+pub const INCR = bindings.INCR;
+pub const DECR = bindings.DECR;
+pub const VENDOR = bindings.VENDOR;
+pub const RENDERER = bindings.RENDERER;
+pub const VERSION = bindings.VERSION;
+pub const EXTENSIONS = bindings.EXTENSIONS;
+pub const NEAREST = bindings.NEAREST;
+pub const LINEAR = bindings.LINEAR;
+pub const NEAREST_MIPMAP_NEAREST = bindings.NEAREST_MIPMAP_NEAREST;
+pub const LINEAR_MIPMAP_NEAREST = bindings.LINEAR_MIPMAP_NEAREST;
+pub const NEAREST_MIPMAP_LINEAR = bindings.NEAREST_MIPMAP_LINEAR;
+pub const LINEAR_MIPMAP_LINEAR = bindings.LINEAR_MIPMAP_LINEAR;
+pub const TEXTURE_MAG_FILTER = bindings.TEXTURE_MAG_FILTER;
+pub const TEXTURE_MIN_FILTER = bindings.TEXTURE_MIN_FILTER;
+pub const TEXTURE_WRAP_S = bindings.TEXTURE_WRAP_S;
+pub const TEXTURE_WRAP_T = bindings.TEXTURE_WRAP_T;
+pub const REPEAT = bindings.REPEAT;
+
+// pub var cullFace: *const fn (mode: Enum) callconv(.C) void = undefined;
+// pub var frontFace: *const fn (mode: Enum) callconv(.C) void = undefined;
+// pub var hint: *const fn (target: Enum, mode: Enum) callconv(.C) void = undefined;
+// pub var lineWidth: *const fn (width: Float) callconv(.C) void = undefined;
+// pub var pointSize: *const fn (size: Float) callconv(.C) void = undefined;
+// pub var polygonMode: *const fn (face: Enum, mode: Enum) callconv(.C) void = undefined;
+// pub var scissor: *const fn (x: Int, y: Int, width: Sizei, height: Sizei) callconv(.C) void = undefined;
+// pub var texParameterf: *const fn (target: Enum, pname: Enum, param: Float) callconv(.C) void = undefined;
+// pub var texParameterfv: *const fn (target: Enum, pname: Enum, params: [*c]const Float) callconv(.C) void = undefined;
+
+// pub var texParameteri: *const fn (target: Enum, pname: Enum, param: Int) callconv(.C) void = undefined;
+pub fn texParameteri(target: TextureTarget, pname: TextureParameter, param: Int) void {
+    bindings.texParameteri(@enumToInt(target), @enumToInt(pname), param);
+}
+
+// pub var texParameteriv: *const fn (target: Enum, pname: Enum, params: [*c]const Int) callconv(.C) void = undefined;
+// pub var texImage1D: *const fn (
+//     target: Enum,
+//     level: Int,
+//     internalformat: Enum,
+//     width: Sizei,
+//     border: Int,
+//     format: Enum,
+//     type: Enum,
+//     pixels: ?*const anyopaque,
+// ) callconv(.C) void = undefined;
+
+// pub var texImage2D: *const fn (
+//     target: Enum,
+//     level: Int,
+//     internalformat: Enum,
+//     width: Sizei,
+//     height: Sizei,
+//     border: Int,
+//     format: Enum,
+//     type: Enum,
+//     pixels: ?*const anyopaque,
+// ) callconv(.C) void = undefined;
+pub fn texImage2D(args: struct {
+    target: TextureTarget,
+    level: usize,
+    internal_format: TextureInternalFormat,
+    width: usize,
+    height: usize,
+    format: PixelFormat,
+    pixel_type: PixelType,
+    data: ?[*]const u8,
```

Comment:
Maybe it should be slice? We could then do an additional check:
```zig
assert(data.size == width * height * format.sizeInBytes());
```

Reply:
Well actually let's leave it as it is for now. Size is specified by `width`, `height` and `pixel_type`. What do you think?

Reply:
I'm not sure if you'd ever want to "underrun" the texture buffer? I think your suggestion makes sense. However we do have to unwrap the optional if it is a slice, like the updated bufferData: https://github.com/michal-z/zig-gamedev/pull/236/commits/9eb073af46a522f6bceb73729c0dbfb2d54e0162